packages/openapi3/test/additional-properties.test.ts

```diff
@@ -0,0 +1,41 @@
+import { ok } from "assert";
+import { diagnoseOpenApiFor, oapiForModel } from "./test-host.js";
+
+describe("openapi3: Additional properties", () => {
```

Comment:
@markcowl can you please review the test cases to make sure I understood them correctly and captured the full scope of the issue.